src/lib/LQCALC/LQCALC.h

```diff
@@ -1,23 +1,90 @@
 #pragma once
 
-#include <stdio.h>
-#include "targets.h"
+#include <stdint.h>
 
+template <uint8_t N>
 class LQCALC
 {
 public:
-    void init(uint8_t depth);
-    void init();
-    void ICACHE_RAM_ATTR add();
-    void ICACHE_RAM_ATTR inc();
-    uint8_t ICACHE_RAM_ATTR getLQ();
-    void ICACHE_RAM_ATTR reset();
-    bool ICACHE_RAM_ATTR packetReceivedForPreviousFrame();
+    LQCALC(void)
+    {
+        reset();
+    }
+
+    /* Set the bit for the current period to true and update the running LQ */
+    void add()
```

Comment:
Don't most of these methods need to be either `inline` or marked `ICACHE_RAM_ATTR` as they are called from ISRs?

Reply:
No, the code is all inserted inline anyway so the decorators were superfluous. If the bodies of the functions were in a separate unit, then it would have to be declared like that, but then it also wouldn't compile as a class template. Correct me if I am wrong. I compiled the code for ESP32, ESP8266, and STM32 (F3) and they all come out the same size and there's no symbol for any LQCALC members, so they've all been inlined.

Reply:
Well blow me over! You are correct. I just compiled with -S and looked at the generated assembly. I was always under the impression that if the class was in a header file then the compiler generated a synthetic function in the object file that included the header. Things have obviously changed/improved in the land of C.

Reply:
It may not always be the case that they're always inlined, but in my experience they are (at least on AVR and STM32). I'm sure that will bite me in the butt somewhere down the line!